clippy_lints/src/methods/mod.rs

```diff
@@ -1553,6 +1585,7 @@ impl<'tcx> LateLintPass<'tcx> for Methods {
             ["get_or_insert_with", ..] => unnecessary_lazy_eval::lint(cx, expr, arg_lists[0], "get_or_insert"),
             ["ok_or_else", ..] => unnecessary_lazy_eval::lint(cx, expr, arg_lists[0], "ok_or"),
             ["collect", "map"] => lint_map_collect(cx, expr, arg_lists[1], arg_lists[0]),
+            ["for_each", "inspect"] => inspect_then_for_each::lint(cx, expr, arg_lists[1], arg_lists[0]),
```

Comment:
To restrict the span to the `inspect` method call, you can pass `method_spans[1]` here and then use `inspect_span.with_hi(expr.span.hi())` as the span in the `span_lint_and_help` function.

Reply:
`inspect_span.with_hi(expr.span.hi())` does the pointing to `inspect` which I couldn't with simply `expr.span`! What does `with_hi` stand for?